addons/l10n_th/models/res_partner.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import models, fields
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    l10n_th_branch_name = fields.Char(compute="_l10n_th_get_branch_name")
+
+    def _l10n_th_get_branch_name(self):
+        for rec in self:
+            if not rec.is_company or rec.country_code != 'TH':
+                rec.l10n_th_branch_name = ""
+            code = self.company_registry
```

Comment:
In Thailand the company_registry field isn't used.
Instead, we're hijacking it to contain the branch name (if the company has several branches).
For example, for the headquarter it will be blank; for the first branch it will be `0001`, for the second `0002` etc.

Reply:
> we're hijacking it

:confused: